config/config_test.go

```diff
@@ -74,6 +73,12 @@ workflows:
 		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			t.Cleanup(func() {
+				cancel()
+				time.Sleep(time.Millisecond)
```

Comment:
this is something we may want to dig into in the future to see why the race-condition happens